enclave/src/cert.rs

```diff
@@ -323,30 +325,14 @@ fn verify_attn_report(report_raw: &[u8], pub_k: Vec<u8>) -> Result<(), sgx_statu
 						);
 					}
 
-					let mut update_info = sgx_update_info_bit_t::default();
-					let mut rt: sgx_status_t = sgx_status_t::SGX_ERROR_UNEXPECTED;
-					let res = unsafe {
-						ocall_get_update_info(
-							&mut rt as *mut sgx_status_t,
-							buf.as_slice().as_ptr() as *const sgx_platform_info_t,
-							1,
-							&mut update_info as *mut sgx_update_info_bit_t,
-						)
-					};
-					if res != sgx_status_t::SGX_SUCCESS {
-						error!("ocall_get_update_info failed. res={:?}", res);
-						return Err(res)
-					}
+					// ArrayVec .into_inner() requires that all elements are occupied by a value
+					// if that's not the case, the following error will occur
+					let platform_info = buf.into_inner().map_err(|e| {
+						error!("Failed to extract platform info from InfoBlob, result does not contain enough elements (require: {}, found: {})", e.capacity(), e.len());
+						sgx_status_t::SGX_ERROR_UNEXPECTED
+					})?;
```

Comment:
Here we use the `ArrayVec` so we can `.push` like a regular vector, but afterwards we can get the inner fixed size array, using `into_inner()`. It has to be noted, that this method checks if all the elements were filled (i.e. `len` == `capacity`)

Reply:
yes push can still panic, that has not changed - I can add the check, using `try_push` .

`into_inner()` exposes the inner fixed size array. I don't think there is another way to access that?

Reply:
I am just wondering, according to our conversation, you seemed to be explicitly looking for something that `Vec` cannot do, but I don't see that here?

I think it has changed, pushing in the standard `Vec` can't panic. It will allocate more storage if the capacity is exceeded.

Reply:
Ah, ok I see it now. You wanted to simplify cloning the data into `platform_info`.

Reply:
yes, that was my motivation to use the `ArrayVec`, so I can easily get the fixed-size array out of it. Although in the end, I think it didn't make a huge difference (because `.push` needs to be checked now).

Reply:
Haha, a classic of the `no-free-food` theorem. In the future, we should probably implement a trait ourselves to solve this in a one-liner.